docusaurus/docs/protocol/tokenomics/token_logic_modules.md

```diff
@@ -0,0 +1,287 @@
+
+## Background: Max Claimable Amount
+
+_tl;dr Max Claimable Amount ∝ (Application Stake / Number of Suppliers per Session)_
+
+Per **Algorithm 1** of the [Relay Mining paper](https://arxiv.org/pdf/2305.10672),
+the maximum amount a supplier can claim from an application in a single session is
+proportional to the Application's stake divided by the number of suppliers in the session.
+
+This is referred to as "Relay Mining Payable Relay Accumulation" in the paper and
+can be seen in the following image:
+
+![Algorithm 1](https://github.com/user-attachments/assets/d1a61535-aa31-447d-88ea-c8d14dcb20c6)
+
+## TLM (pre) Processing
+
+_tl;dr Determine if the claim settlement amount is greater than the maximum claimable amount and then run each individual TLM._
+
+**Prior to** processing each individual TLM, we need to understand if the amount claimed
+by the supplier adheres to the optimistic maxima set per the limits of Relay Mining.
+
+Suppliers always have the option to over-service an application (**i.e. do free work**),
+in exchange for providing a good service to the network. This may lead to offchain
+reputation benefits (e.g. Gateways favoring them), but their on-chain rewards
+are still limited as a function of the application's stake before the session started
+and the number of suppliers in the session.
+
+:::note
+
+TODO_POST_MAINNET: After the Shannon upgrade, the team at Grove has a lot of ideas
+related to on-chain reputation, [supplier gossiping](https://www.notion.so/buildwithgrove/Off-chain-Application-Stake-Tracking-6a8bebb107db4f7f9dc62cbe7ba555f7?pvs=4), and
+much more, but that is out of scope for the initial implementation.
+
+:::
+
+```mermaid
+---
+title: "Token Logic Modules (pre) Processing"
+---
+flowchart TB
+    CSA(["Claim Settlement Amount (CSA)"])
+    MCA(["MaxClaimPerSupplier (MCA) <br> = (AppStake / NumSuppliersPerSession)"])
+    CC{"Is CSA > MCA?"}
+    Update(Set SA = MCA <br>Broadcast Event)
+    SOAE{{Application Overserviced <br> Event}}
+    TLMP[["TLM Processor (SA)"]]
+
+    CSA -- CSA --> CC
+    MCA -- MCA --> CC
+
+    Update -..-> SOAE
+    CC -- Yes --> Update
+    CC -- No<br>SA=CSA --> TLMP
+    Update -- SA=CSA --> TLMP
+
+    TLMP --SA--> TLMBEM[[TLM: Burn Equals Mint]]
+    TLMP --SA--> TLMGI[[TLM: Global Inflation]]
+    TLMP --SA--> TLMGIRR[[TLM: Global Inflation Reimbursement Request]]
+
+    classDef tlm fill:#54ebd5,stroke:#333,stroke-width:2px;
+    classDef question fill:#e3db6d,stroke:#333,stroke-width:2px;
+    classDef event fill:#e8b761,stroke:#333,stroke-width:2px;
+
+    class TLMP,TLMBEM,TLMGI,TLMGIRR tlm;
+    class SOAE event;
+    class CC question;
+```
+
+## TLM: Mint=Burn (MEB)
+
+_tl;dr The transfer of tokens from the application to the supplier based on the amount of work received and provided respectively.._
+
+The `Mint=Burn` TLM is, _theoretically_, the only TLM necessary when the network
+reaches equilibrium in the far future.
```

Comment:
What's "reaching equilibrium", supply/demand?

Reply:
`reaches maturity in the far future.`